script.js

```diff
@@ -1,4 +1,208 @@
+  for (let i = 1; i <= 13; i += 1) {
+    var suits = ["♦", "♥", "♣", "♠"];
+    for (let j = 0; j < suits.length; j += 1) {
+      var name = `${i}`;
+      if (name === "11") {
+        name = "Jack";
+      } else if (name === "12") {
+        name = "Queen";
+      } else if (name === "13") {
+        name = "King";
+      }
+      var value = parseInt(name);
+      if (name === "Jack" || name === "Queen" || name === "King") {
+        value = 10;
+      }
+
+      var card = {
+        value: value,
+        suit: suits[j],
+        img: imgholder[i - 1][j],
+        name,
+      };
+      newDeck.push(card);
+    }
+  }
+  return newDeck;
 };
+// Get a random index ranging from 0 (inclusive) to max (exclusive).
+const getRandomIndex = (max) => Math.floor(Math.random() * max);
+// Shuffle an array of cards
+const shuffleCards = (deck) => {
+  // Loop over the card deck array once
+  for (let currentIndex = 0; currentIndex < deck.length; currentIndex += 1) {
+    // Select a random index in the deck
+    const randomIndex = getRandomIndex(deck.length);
+    // Select the card that corresponds to randomIndex
+    const randomCard = deck[randomIndex];
+    // Select the card that corresponds to currentIndex
+    const currentCard = deck[currentIndex];
+    // Swap positions of randomCard and currentCard in the deck
+    deck[currentIndex] = randomCard;
+    deck[randomIndex] = currentCard;
+  }
+  // Return the shuffled deck
+  return deck;
+};
+
+//define global variables
+var playerCardPlaceholders = document.getElementsByClassName("p-card");
+var computerCardPlaceholders = document.getElementsByClassName("c-card");
+var deck = makeDeck();
+var shuffledDeck = shuffleCards(deck);
+var playButton = document.getElementById("play-button");
+var hitButton = document.getElementById("hit-button");
+var standButton = document.getElementById("stand-button");
+var restartButton = document.getElementById("restart-button");
+var outputMessageElement = document.getElementById("output");
+
+var playerCards = [];
+var computerCards = [];
+var playerScore = 0;
+var computerScore = 0;
+
+//defind helper functions
+function checkBlackJacket(num1, num2) {
+  if ((num1 == 1 && num2 == 10) || (num1 == 10 && num2 == 1)) {
+    score = 21;
+  } else {
+    score = 0;
+  }
+  return score;
+}
+
+function compareScore(num1, num2) {
+  var winner;
+  if ((num1 == 21 && num2 == 21) || (num1 > 21 && num2 > 21)) {
+    winner = "It's a draw!";
+  } else if (num1 < 21 && num2 < 21) {
+    if (21 - num1 < 21 - num2) {
+      winner = "The winner is Player.";
+    } else if (21 - num1 > 21 - num2) {
+      winner = "The winner is Computer.";
+    } else {
+      winner = "It's a Draw!";
+    }
+  } else if (num1 > 21 && num2 <= 21) {
+    winner = "The winner is Computer.";
+  } else if (num2 > 21 && num1 <= 21) {
+    winner = "The winner is Player.";
+  }
+  return winner;
+}
+
+playButton.addEventListener("click", () => {
+  for (var i = 0; i < 2; i++) {
```

Comment:
the next 3 lines seem to be repeated at a couple of places. It's a hint that you probably want a function for it